src/ios/THEAppContext.m

```diff
@@ -74,611 +70,311 @@ - (void)applicationDidBecomeActiveNotification:(NSNotification *)notification;
-                                               peerName:params[1]];
-            [JXcore callEventCallback:callbackId
-                           withParams:@[@(true)]];
-        }
-    } withName:@"StartPeerCommunications"];
-    
-    // StopPeerCommunications native block.
-    [JXcore addNativeBlock:^(NSArray * params, NSString * callbackId) {
-        [self stopCommunications];
-        [JXcore callEventCallback:callbackId
-                       withParams:nil];
-    } withName:@"StopPeerCommunications"];
-
-    // BeginConnectToPeerServer native block.
-    [JXcore addNativeBlock:^(NSArray * params, NSString * callbackId) {
-        // Obtain the peer identifier.
-        NSString * peerIdentifier = params[0];
-        
-        // Connect to the the peer server.
-        BOOL result = [self connectToPeerServerWithPeerIdentifier:[[NSUUID alloc] initWithUUIDString:peerIdentifier]];
-        
-        // Return the result.
-        [JXcore callEventCallback:callbackId
-                       withParams:@[@(result)]];
-    } withName:@"BeginConnectToPeerServer"];
-    
-    // DisconnectPeerServer native block.
-    [JXcore addNativeBlock:^(NSArray * params, NSString * callbackId) {
-        // Obtain the peer identifier.
-        NSString * peerIdentifier = params[0];
-        
-        // Disconnect from the peer server.
-        BOOL result = [self disconnectFromPeerServerWithPeerIdentifier:[[NSUUID alloc] initWithUUIDString:peerIdentifier]];
-        
-        // Return the result.
-        [JXcore callEventCallback:callbackId
-                       withParams:@[@(result)]];
-    } withName:@"DisconnectFromPeerServer"];
-}
 
 // Starts communications.
-- (void)startCommunicationsWithPeerIdentifier:(NSUUID *)peerIdentifier
-                                     peerName:(NSString *)peerName
+- (BOOL)startBroadcasting:(NSString *)peerIdentifier serverPort:(NSNumber *)serverPort
 {
-    if ([_atomicFlagCommunicationsEnabled trySet])
-    {
-        // Allocate and initialize the service type.
-        NSUUID * serviceType = [[NSUUID alloc] initWithUUIDString:@"72D83A8B-9BE7-474B-8D2E-556653063A5B"];
-
-        // Allocate and initialize the peer Bluetooth context.
-        _peerBluetooth = [[THEPeerBluetooth alloc] initWithServiceType:serviceType
-                                                        peerIdentifier:peerIdentifier
-                                                              peerName:peerName];
-        [_peerBluetooth setDelegate:(id<THEPeerBluetoothDelegate>)self];
-        
-        // Allocate and initialize peer networking.
-        _peerNetworking = [[THEPeerNetworking alloc] initWithServiceType:@"Thali"
-                                                          peerIdentifier:peerIdentifier
-                                                                peerName:peerName];
-        [_peerNetworking setDelegate:(id<THEPeerNetworkingDelegate>)self];
+  if ([_atomicFlagCommunicationsEnabled trySet])
```

Comment:
Do we really need an atomic guard here? Shouldn't startBroadcasting only be called by JXcore and doesn't JXcore guarantee that it will always call this object on the same thread? In which case isn't it literally impossible to have two calls to startBroadcasting simultaneously? If true wouldn't this apply to all uses of atomicFlagcommunicationsEnabled in this file?